association-backend/src/webhooks/handlers/crons/membership_check_status.py

```diff
@@ -0,0 +1,64 @@
+import logging
+from datetime import datetime, timezone
+from typing import Any
+
+from src.association_membership.domain.entities import (
+    PaymentStatus,
+    Subscription,
+    SubscriptionStatus,
+)
+from src.association_membership.domain.repository import AssociationMembershipRepository
+
+logger = logging.getLogger(__file__)
+
+
+async def membership_check_status(payload: Any):
+    await update_expired_subscriptions()
+    await update_now_active_subscriptions()
+
+
+async def update_now_active_subscriptions():
+    repository = AssociationMembershipRepository()
+
+    now = datetime.now(timezone.utc)
+    # Ideally in the future we should use the psql NOW() function
+    qs = Subscription.objects.filter(
+        status=SubscriptionStatus.CANCELED,
+        payments__status=PaymentStatus.PAID,
+        payments__period_start__lte=now,
+        payments__period_end__gte=now,
+    )
+    subscriptions_to_enable = await qs.all()
+    subscriptions_to_enable_count = await qs.count()
+
+    logger.info(
+        "Found subscriptions_to_enable_count=%s subscriptions to activate",
+        subscriptions_to_enable_count,
+    )
+
+    for subscription in subscriptions_to_enable:
+        subscription.mark_as_active()
+        await repository.save_subscription(subscription)
+
+
+async def update_expired_subscriptions():
+    repository = AssociationMembershipRepository()
+
+    now = datetime.now(timezone.utc)
+    # Ideally in the future we should use the psql NOW() function
+    qs = Subscription.objects.filter(status=SubscriptionStatus.ACTIVE,).exclude(
+        payments__status=PaymentStatus.PAID,
+        payments__period_start__lte=now,
+        payments__period_end__gte=now,
```

Comment:
But if the payment end period is Greater Than or Equal now, doesn't mean it's still active??

Reply:
Yeah, that's why it is inside the `exclude`

```
qs = Subscription.objects.filter(status=SubscriptionStatus.ACTIVE,).exclude(
```

so we exclude subscriptions that are active

Reply:
so period_start should be in the past (lte)
and period_end should be in the future (gte)

so the subscription is still covered, exclude it